### PR TITLE
SNP term type improvements

### PR DESCRIPTION
**Pull request by gavrielm**
## Description

Support query for snp term type by position, dbsnp id, and variant format.
Support groupsetting edit menu for snp term type.
Support snp term type in cumulative incidence analysis.
Fixed https://github.com/stjude/proteinpaint/issues/1910

To test:

1. Open [sjlife portal](http://localhost:3000/?noheader=1&mass={%22genome%22:%22hg38%22,%22dslabel%22:%22SJLife%22})
2. Open data dictionary -> SNP Genotype tab -> search for SNP by:
- dbsnp id (`rs1641548`)
- position (`chr17:7671461` or `chr17:7671461-7671462` or `chr17:7671461-7671463`)
- variant format (`chr17.7671461.C.T`)

3. Test snp term type edit menu
4. Create any cumulative incidence plot and overlay a snp term type

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

**Review by siosonel — CHANGES_REQUESTED** [review]
Major issue:
- I overlayed rs1641548 on a arrhythmias cuminc plot, it worked. I edited to create two groups (separate C/C), the loading hangs, the[ server responds with an error](http://localhost:3000/termdb?genome=hg38&dslabel=SJLife&embedder=localhost&getcuminc=1&term1_$id=ZMSmX1N577Sflm%2Fac09ptCqYxMg-&term1_id=Arrhythmias&term1_q=%7B%22isAtomic%22%3Atrue%2C%22hiddenValues%22%3A%7B%22-1%22%3A1%7D%2C%22mode%22%3A%22cuminc%22%2C%22value_by_max_grade%22%3Atrue%2C%22bar_by_grade%22%3Atrue%2C%22breaks%22%3A%5B3%5D%7D&term2=%7B%22id%22%3A%22rs1641548%22%2C%22chr%22%3A%22chr17%22%2C%22start%22%3A7671460%2C%22stop%22%3A7671461%2C%22name%22%3A%22rs1641548%22%2C%22ref%22%3A%22C%22%2C%22alt%22%3A%5B%22T%22%5D%2C%22type%22%3A%22snp%22%7D&term2_q=%7B%22isAtomic%22%3Atrue%2C%22type%22%3A%22custom-groupset%22%2C%22groupsetting%22%3A%7B%22inuse%22%3Atrue%2C%22customset%22%3A%7B%22groups%22%3A%5B%7B%22name%22%3A%22Group%201%22%2C%22type%22%3A%22values%22%2C%22values%22%3A%5B%7B%22key%22%3A%22T%2FT%22%2C%22label%22%3A%22T%2FT%22%7D%2C%7B%22key%22%3A%22C%2FT%22%2C%22label%22%3A%22C%2FT%22%7D%5D%7D%2C%7B%22name%22%3A%22Group%202%22%2C%22type%22%3A%22values%22%2C%22values%22%3A%5B%7B%22key%22%3A%22C%2FC%22%2C%22label%22%3A%22C%2FC%22%7D%5D%7D%5D%7D%7D%7D&filter=%7B%22type%22%3A%22tvslst%22%2C%22in%22%3Atrue%2C%22join%22%3A%22%22%2C%22lst%22%3A%5B%7B%22tag%22%3A%22cohortFilter%22%2C%22type%22%3A%22tvs%22%2C%22tvs%22%3A%7B%22term%22%3A%7B%22id%22%3A%22subcohort%22%2C%22type%22%3A%22categorical%22%7D%2C%22values%22%3A%5B%7B%22key%22%3A%22SJLIFE%22%2C%22label%22%3A%22SJLIFE%22%7D%5D%7D%7D%5D%7D&minSampleSize=10&hiddenValues=%7B%22term1%22%3A%5B%22Not%20tested%22%5D%2C%22term2%22%3A%5B%5D%7D) "Cannot read properties of undefined (reading 'key')"

Minor issues
- using the variant format chr17.7671461.C.T, there are no displayed matching search results, but hitting enter works
- at the snp termsetting, after creating groupsetting and applied to chart, edit the termsetting again by changing the group names doesn't work

File: client/dom/genesearch.ts [hunk 1]
```diff
@@ -133,6 +134,11 @@ export function addGeneSearchbox(arg: GeneSearchBoxArg) {
 	const tip = arg.tip,
 		row = arg.row
 
+	if (arg.geneOnly && arg.snpOnly) {
+		row.append('span').text('.geneOnly and snpOnly cannot both be true')
```
Comment (siosonel) [inline client/dom/genesearch.ts:138]: Not a good use of boolean flags, should be converted to maybe `arg.restriction: 'gene' | 'snp' | 'none'``. The boolean flag requires validation whereas using the same arg attribute would not require validation. The boolean flag also will require a more  complicated or less intuitive type definition.

Comment (xzhou82) [reply client/dom/genesearch.ts:138]: yes will do this refactor later

**Comment by gavrielm** [comment]
I fixed the issue with editing a snp term in the cuminc plot. This issue is related to a broader issue with `tw.$id` becoming `undefined` when a term is edited (dictionary or non-dictionary terms). This should be addressed in the next PR.

**Comment by xzhou82** [comment]
the groupsetting group name change not working issue also exist in master for categorical term. i agree to fix it in next pr

**Review by xzhou82 — APPROVED** [review]
all CI pass, thanks

**Comment by gavrielm** [comment]
Some typescript errors need to be fixed

**Review by xzhou82 — APPROVED** [review]

**Review by siosonel — APPROVED** [review]

**Pull request merged**